server/__tests__/signin.spec.js

```diff
@@ -0,0 +1,117 @@
+import chai, { expect } from 'chai';
+import chaiHttp from 'chai-http';
+import app from '../../app';
+import baseUrl from './utils/baseUrl.utils';
+import mock from './utils/mock';
+
+chai.use(chaiHttp);
+
+describe('User Sign In', () => {
+  it('should successfully sign in an existing user', async () => {
+    const res = await chai
+      .request(app)
+      .post(`${baseUrl}/auth/signin`)
+      .send({
+        ...mock.signin
+      });
+    expect(res.status).to.equal(200);
+    expect(res.body.message).to.equal('Successfully signed in');
+    expect(res.body.data[0].user.email).to.equal(mock.signin.email);
+  });
+
+  it('should not signin a non-existing email', async () => {
+    const res = await chai
+      .request(app)
+      .post(`${baseUrl}/auth/signin`)
+      .send({
+        ...mock.signin,
+        email: 'holly@holly.com',
+      });
+    expect(res.status).to.equal(404);
```

Comment:
I think a `404` is not ideal for `invalid login credentials`. `400` might be more ideal.

Reply:
Thanks....will make the changes``